src/main/java/com/wire/bots/hold/utils/Cache.java

```diff
@@ -12,9 +13,11 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 public class Cache {
-    private static final ConcurrentHashMap<UUID, File> assets = new ConcurrentHashMap<>();//<messageId, File>
-    private static final ConcurrentHashMap<UUID, User> users = new ConcurrentHashMap<>();//<userId, User>
-    private static final ConcurrentHashMap<UUID, File> profiles = new ConcurrentHashMap<>();//<userId, Picture>
+    private static String DEFAULT_DOMAIN = null;
```

Comment:
I don't have enough context, but it seems to me that this is an in-memory cache? Or is it persisted?

If it's in-memory only, perhaps the default domain value should be persisted instead, and perhaps the app should throw a fatal  error if it has previously stored data (users, conversations, ...) with one domain, but at start up checks the domain with the backend and it's now aa different domain.

Otherwise the code might make a lot of wrong assumptions when looking at old local data. If the data (users, conversations, ...) was stored with `null` domain when the default domain was A, but now the defalt domain is B, the data will be inconsistent. 

Or do we always store the domain explicitly in the data that is persisted?

Reply:
Yes, it is only stored in-memory.

You are correct in the case of the application being shutdown and then changing the domain, when starting again it will assume a different domain and will have inconsistent data (domain) between values that was previously stored.

If that is an issue, then we can persist the default domain in the database and throw a fatal error if when restarting the service and the fetched domain from `/api-version` is different from the one persisted, then we can throw a fatal error.

Should we have a quick meeting between me, you and @spoonman01 about the final approach? or can we continue with what was said:

- If in DB default domain is null, fetch from `/api-version` and persist it.
- if in DB default domain is NOT null, when fetching from `/api-version` and the values are different, we throw a fatal exception.

Reply:
To be honest, this situation (the domain changes) is extremely unlikely, but I like to do a bit of defensive programming. So we should look at this with a risk-cost analysis. Is the change to consider this scenrio a substantial change that will take a while to implement? Then perhaps we accept the risk. Is it relatively easy to implement? Then I would implement it.

Reply:
We created a new table called `Metadata` that holds 2 fields :`key` and `value` (so it can also be used later if we need more metadata/configs to be saved)

And also created a file called `FallbackDomainFetcher` where it runs on service start to verify if it contains a value in Cache, otherwise fetch from the API and compare against what is saved in the database (if there is any domain saved).

In case the domain received from the API is different from the one in the database, then we are throwing a `RuntimeException`.